piptools/sync.py

```diff
@@ -157,26 +164,34 @@ def sync(to_install, to_uninstall, verbose=False, dry_run=False, install_flags=N
     if not verbose:
         pip_flags += ["-q"]
 
-    if to_uninstall:
-        if dry_run:
+    if ask:
+        dry_run = True
```

Comment:
I think this `if` statement should be removed in favor of checking the `ask` below.